app/main.py

```diff
@@ -1,25 +1,131 @@
 from abc import ABC
+from typing import Any
 
 
 class IntegerRange:
-    pass
+    def __init__(
+            self,
+            min_amount: int,
+            max_amount: int
+    ) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __get__(
+            self,
+            obj: Any,
+            objtype: None | Any = None
+    ) -> int | str:
+        if obj is None:
+            return self
```

Comment:
There is a mistake here. The `__get__` method should return the value of the attribute, not `self`. Returning `self` is incorrect and will lead to unexpected behavior when accessing the attribute.